src/opensearch.ts

```diff
@@ -0,0 +1,113 @@
+import { ApiResponse, Client } from '@opensearch-project/opensearch'
+import { TransportRequestPromise } from '@opensearch-project/opensearch/lib/Transport'
+import { taskEither } from 'fp-ts'
+import { pipe } from 'fp-ts/lib/pipeable'
+import { chain, TaskEither, tryCatch } from 'fp-ts/lib/TaskEither'
+import { extractSource } from './extract'
+import { ApplicationError, StatusCodes } from './http'
+
+enum OSIndex {
+    Snapshots = 'snapshots_kris_1'
+}
+
+export const getClient = (): Client => {
+    return new Client({ node: 'http://localhost:9501' })
+}
+
+export const getNestedObject = (
+    path: string,
+    term: string,
+    value: string,
+    includes: string[]
+) => {
+    return {
+        path: path,
+        query: {
+            bool: {
+                filter: {
+                    term: {
+                        [path + '.' + term]: {
+                            value
+                        }
+                    }
+                }
+            }
+        },
+        ignore_unmapped: false,
+        inner_hits: {
+            ignore_unmapped: false,
+            from: 0,
+            size: 7,
+            version: false,
+            seq_no_primary_term: false,
+            explain: false,
+            track_scores: false,
+            _source: {
+                includes: includes.map(name => path + '.' + name)
+            }
+        }
+    }
+}
+
+export const getBalanceByAddress = (es: Client) => (address: string, ordinal: string = 'latest'): TaskEither<ApplicationError, any> => {
+
+    const innerPath = 'info.balances'
+    const nested = getNestedObject(innerPath, 'address', address, ['balance'])
+    const osSearch = getByFieldQuery(OSIndex.Snapshots, 'ordinal.value', ordinal, ['ordinal'], nested)(es)
+
+    return findOne(innerPath)(osSearch)
+}
+
+const getByFieldQuery = (
+    index: string,
+    field: string,
+    value: string,
+    includes: string[] = [],
+    nestedBody: any,
+    size: number | null = 1
+) => (es: Client) =>
+        es.search({
+            index,
+            body: {
+                size,
+                query: {
+                    bool: {
+                        filter: [
+                            {
+                                term: {
+                                    [field]: {
+                                        value: value
+                                    }
+                                }
+                            },
+                            { nested: nestedBody }
+                        ]
+                    }
+                },
+                _source: {
+                    includes: includes
+                }
+            }
+        })
+
+const concatResults = (res1: any[]) => (res2: any[]) => Object.assign(res1[0], res2[0])
```

Comment:
Are you sure that these arrays are not empty?

Reply:
assuming you're concatenating the same type, I'd write it like that:
```typescript
const concatResults<A> = (res1: A[]) => (res2: A[]): A[] => [...res1, ...res2]
```
but I believe it's easier to use `[...a, ...b]` instead of creating an aux function to do that 🤔